pkg/operators/limiter/limiter.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package limiter is a data operator that operates on array data sources by
+// limiting the number of entries in each batch of data. A great scenario for
+// this operator is when you are already sorting data within an array of data
+// and you want to filter out the top `X` entries.
+package limiter
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+const (
+	name            = "limiter"
+	ParamMaxEntries = "max-entries"
+	Priority        = 9600
+)
+
+type limiterOperator struct{}
+
+func (l *limiterOperator) Name() string {
+	return name
+}
+
+func (l *limiterOperator) Init(params *params.Params) error {
+	return nil
+}
+
+func (l *limiterOperator) GlobalParams() api.Params {
+	return nil
+}
+
+func (l *limiterOperator) InstanceParams() api.Params {
+	return api.Params{
+		{
+			Key:   ParamMaxEntries,
+			Title: "Max Entries",
+			Description: "The maximal number of entries for each batch of data. " +
+				"If using multiple data sources, prefix the value with 'datasourcename:' and separate with ';'.",
+			DefaultValue: "50",
+			TypeHint:     api.TypeString,
+		},
+	}
+}
+
+func (l *limiterOperator) InstantiateDataOperator(gadgetCtx operators.GadgetContext, instanceParamValues api.ParamValues) (operators.DataOperatorInstance, error) {
+	for _, ds := range gadgetCtx.GetDataSources() {
+		if ds.Type() == datasource.TypeArray {
+			return &limiterOperatorInstance{
+				maxEntries: instanceParamValues[ParamMaxEntries],
+			}, nil
```

Comment:
This creates a `limiterOperatorInstance` even if there's no param value set. Maybe add a check? (I know you're providing a default above, so it should be filled in from the clients using that value, but using the API or other ways, it might be empty which would lead to no returned value (as default would be 0, right?))

Reply:
I tried to parse `max-entries` in `InstantiateDataOperator` and instantiate it only if `max-entries` has a valid value, but I found an issue doing so. 

`InstantiateDataOperator` is also called during the `GetGadgetInfo` and, in that moment, the `instanceParamValues` are not yet available. So, we'd wrongly conclude "we don't need the limiter, so don't instantiate it", resulting in not including the operator's params in the `GadgetInfo`.

Sometimes I think that the instantiation for `GetGadgetInfo` should be different than the one for `Run`.

Reply:
Ah, right - that happened for other operators as well, already. We could still check for errors and in any case return at least an empty operator if there's an array ds to make sure params are transported.

> Sometimes I think that the instantiation for GetGadgetInfo should be different than the one for Run.

Would make it even more complicated IMHO. But we should build a list of things to do / not to do in initialization & so on :)

Reply:
> We could still check for errors and in any case return at least an empty operator if there's an array ds to make sure params are transported.
>
I followed this suggestion to detect wrong flag values earlier.